solidity/contracts/BancorConverterUpgrader.sol

```diff
@@ -0,0 +1,241 @@
+    function getQuickBuyPathLength() public view returns (uint256);
+    function transferTokenOwnership(address _newOwner) public view;
+    function withdrawTokens(IERC20Token _token, address _to, uint256 _amount) public view;
+    function acceptTokenOwnership() public view;
+    function transferManagement(address _newManager) public view;
+    function setConversionFee(uint32 _conversionFee) public view;
+    function setQuickBuyPath(IERC20Token[] _path) public view;
+    function addConnector(IERC20Token _token, uint32 _weight, bool _enableVirtualBalance) public view;
+    function getConnectorBalance(IERC20Token _connectorToken) public view returns (uint256);
+    function getReserveBalance(IERC20Token _reserveToken) public view returns (uint256);
+    function connectors(address _address) public view returns (
+        uint256 virtualBalance, 
+        uint32 weight, 
+        bool isVirtualBalanceEnabled, 
+        bool isPurchaseEnabled, 
+        bool isSet
+    );
+    function reserves(address _address) public view returns (
+        uint256 virtualBalance, 
+        uint32 weight, 
+        bool isVirtualBalanceEnabled, 
+        bool isPurchaseEnabled, 
+        bool isSet
+    );
+}
+
+/*
+    Bancor Upgrade Converter
+
+    The Bancor upgrade converter contract allows converter upgrade from old versions (0.4 and up) to the latest version.
+    Contract deployment proccess will be without arguments.
+    For begining upgrading a converter with contract instance on the network, first transfer the ownership to
+    the upgrade contract instance address and then call to upgradeConverter function.
+    Ownership of the current converter will be transfered back to the given new owner as the ownership of the new converter.
+    You can find the address of the new converter at ConverterCreated event.
+
+    WARNING: The contract assumes that it owned the current converter.
+*/
+contract BancorConverterUpgrader is Owned {
+
+    // triggered when the contract accept a converter ownership
+    event ConverterOwned(address indexed _converter, address indexed _owner);
+    // triggered when the upgrading process is done
+    event ConverterCreated(address indexed _fromConverter, address indexed _toConverter);
+
+    /**
+        @dev constructor
+    */
+    function BancorConverterUpgrader()
+        public
+    {}
+
+    /**
+        @dev upgrade from old converter versions to the latest version
+        will throw if transfer ownership request was not sent before calling this function.
+        ownership of the current and new converter will be transfered to a given owner address.
+        when process is finish the function fired an event.
+
+        @param _fromConverter   converter to upgrade
+        @param _version         current conveter version
+    */
+    function upgradeConverter(IBancorConverter _fromConverter, bytes32 _version)
+        public
+    {
+        bool formerVersions = false;
```

Comment:
no check for original owner, seems like anyone can call upgrade once the ownership transfer has been initiated

Reply:
the check is in acceptConverterOwnership